sync/src/utils.rs

```diff
@@ -23,17 +24,57 @@ pub(crate) fn send_message<Message: Entity>(
     }
 
     let bytes = message.as_bytes().len() as u64;
-    let item_id = item_id(protocol_id, message);
-    metrics!(
-        counter,
-        "ckb.messages_bytes", bytes,
-        "direction" => "out",
-        "protocol_id" => protocol_id.value().to_string(),
-        "item_id" => item_id.to_string(),
-    );
+    let item_name = item_name(protocol_id, message);
+
+    let protocol_name: String = {
+        /// protocol_id to name
+        "protocol id to name".to_string()
+    };
+
+    if ckb_metrics::metrics_enabled() {
+        metric_ckb_message_bytes(
+            MetricDirection::Out,
+            &protocol_name,
+            &item_name,
+            None,
+            bytes,
+        );
+    }
```

Comment:
I want to create a function to get protocol_name from protocol_id.
```rust
/// Convert protocol id to protocol name
/// Will return "Unknown" if the protocol id is not supported in SupportProtocols
pub fn protocol_name(protocol_id: ProtocolId) -> String {
    match protocol_id.value() {
        0 => SupportProtocols::Ping.name(),
        1 => SupportProtocols::Discovery.name(),
        2 => SupportProtocols::Identify.name(),
        3 => SupportProtocols::Feeler.name(),
        4 => SupportProtocols::DisconnectMessage.name(),
        100 => SupportProtocols::Sync.name(),
        101 => SupportProtocols::RelayV2.name(),
        102 => SupportProtocols::Time.name(),
        110 => SupportProtocols::Alert.name(),
        120 => SupportProtocols::LightClient.name(),
        121 => SupportProtocols::Filter.name(),
        _ => "unknown".to_string(),
    }
}
```
Do you think there is a simpler way to achieve this? @driftluo

Reply:
I think if there is an unknown, it is a bug

Reply:
When `fn protocol_message_name()` found a `ProtocolId` which can't be mapped to `SupportedProtocols`, I will return "Unknown" and print a error log:
```rust
fn protocol_message_name<Message: Entity>(
    protocol_id: ProtocolId,
    message: &Message,
) -> (String, String) {
    if protocol_id == SupportProtocols::Sync.protocol_id() {
        (
            SupportProtocols::Sync.name(),
            SyncMessageReader::new_unchecked(message.as_slice())
                .to_enum()
                .item_name()
                .to_owned(),
        )
    } else if protocol_id == SupportProtocols::RelayV2.protocol_id() {
        (
            SupportProtocols::RelayV2.name(),
            RelayMessageReader::new_unchecked(message.as_slice())
                .to_enum()
                .item_name()
                .to_owned(),
        )
    } else {
        error!("send message got an Unknown protocol id: {}", protocol_id);
        ("Unknown".to_string(), Message::NAME.to_owned())
    }
}
```